ValidationSummary doesn't display property errors if ModelState and M…

kishanAnem opened the pull request:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

Hi @dougbu 
#4831
I think this is fixed.

pranavkm committed:
[Design]: Introduce CircuitHandler to handle circuit lifetime events

Partial fix to https://github.com/aspnet/AspNetCore/issues/6353

pranavkm committed:
Fix names and grammar

pranavkm committed:
More PR changes

pranavkm committed:
Remove unused

dougbu committed:
Remove benchmark apps from Mvc.sln
- follow up to 61bc18fee3

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

dougbu reviewed (changes requested):
Also needs a test in `HtmlHelperValidationSummaryTest` for the specific scenario that failed in #4831.

Best if that test does _not_ involve a naming conflict between a parameter and a top-level model property. Add errors for the parameter, the property, and one more that matches neither. (Naming conflicts are a user error and this issue can occur without one.)

SteveSandersonMS committed:
Strong-name the dotnet-blazor CLI tool. Fixes #1501

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

Hi @dougbu

Test case added.

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

Hi @dougbu

Test case added.

SteveSandersonMS committed:
In Blazor Hosted template, use Json.NET. Fixes #7122

SteveSandersonMS committed:
In Blazor templates, update survey links. Fixes #7130

SteveSandersonMS committed:
Update Mono binaries. Fixes #7124

dougbu committed:
Add component detection step to builds
- aspnet/AspNetCore-Internal#1254

pranavkm committed:
Changes per PR

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -587,7 +588,70 @@ public async Task ProcessAsync_GeneratesValidationSummaryWhenNotNone(ValidationS
 
-        private static ViewContext CreateViewContext()
+        [Fact]
+        public async Task ProcessAsync_GeneratesExpectedOutput_WithModelErrorForIEnumerable()
```
With only one error in `ModelState`, I doubt this test will fail without your fix. Please locally revert your changes to `ValidationHelpers` and test with
1. an _error_ for `FormMetadata.ID` (key is likely "ID" aka `nameof(FormMetadata.ID)`)
  - should be duplicated without your fix
2. that error and one for (say) `parameter`
  - should see just the error for `FormMetadata.ID` (twice) without your fix

Then, reapply your fix and confirm both tests display the expected errors without duplication.

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -587,7 +588,70 @@ public async Task ProcessAsync_GeneratesValidationSummaryWhenNotNone(ValidationS
+            validationSummaryTagHelper.ViewContext = viewContext;
+
+            var modelState = viewContext.ModelState;
+            SetValidModelState(modelState);
```
Remove this call. The tag helper doesn't look at the values in ModelState, just the errors.

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -612,6 +676,7 @@ private static void SetValidModelState(ModelStateDictionary modelState)
             modelState.SetModelValue(key: $"{nameof(Model.Strings)}[1]", rawValue: null, attemptedValue: null);
             modelState.SetModelValue(key: $"{nameof(Model.Strings)}[2]", rawValue: null, attemptedValue: null);
             modelState.SetModelValue(key: nameof(Model.Text), rawValue: null, attemptedValue: null);
+            modelState.SetModelValue(key: nameof(FormMetadata.ID), rawValue: null, attemptedValue: null);
```
Undo this change.

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -587,7 +588,70 @@ public async Task ProcessAsync_GeneratesValidationSummaryWhenNotNone(ValidationS
+            Assert.Equal(expectedTagName, output.TagName);
+        }
+
+            private static ViewContext CreateViewContext()
```
Undo the extra indentation here.

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -2,9 +2,10 @@
 using System.Collections.Generic;
 using System.IO;
-using System.Threading.Tasks;
+using System.Threading.Tasks;   
```
Undo this change.

dougbu reviewed (changes requested):
Looks good but needs a few touch-ups and a bit more work on the test(s)

pranavkm committed:
Introduce CircuitHandler to handle circuit lifetime events (#6971)

Introduce CircuitHandler to handle circuit lifetime events

Partial fix to https://github.com/aspnet/AspNetCore/issues/6353

pakrym committed:
More logs in forwardinghandler.cpp (#7244)

jkotalik committed:
Upload artifact logs for CI Test runs (#7251)

jkotalik committed:
Make GetMemory use MaxBufferSize for MemoryPool (#7143)

dougbu committed:
Kill processes after completing build and before publishing test results or artifacts (#7228)

- aspnet/AspNetCore-Internal#1735
- port aspnet/Extensions@e2bb8ac1a0 fix into this repo
  - change `default-build.yml` instead of `azure-pipelines.yml`
  - make a couple of changes to `KillProcesses.ps1` (which I'll take back to Extensions)
- remove `ci-official.yml`

dougbu committed:
Merge branch 'release/2.1' into release/2.2

mikaelm12 committed:
Reduce number of SauceLabs browsers (#7152)

dougbu committed:
Merge branch 'release/2.2'

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

Touch ups done

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and ModelMetadata meets the specific conditions #4831

Touch ups done

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -619,6 +679,27 @@ private static void SetValidModelState(ModelStateDictionary modelState)
             }
         }
 
+        public static TheoryData<ModelStateDictionary, string> ProcessAsync_GeneratesExpectedOutput_WithErrorsData
```
Please move this method just above `ProcessAsync_GeneratesExpectedOutput_WithModelErrorForIEnumerable(...)`. (We usually only have support methods and private classes at the bottom of test classes.)

dougbu commented on src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs:
```diff
@@ -619,6 +679,27 @@ private static void SetValidModelState(ModelStateDictionary modelState)
+                var modelStateError = new ModelStateDictionary();
+                modelStateError.SetModelValue(key: parameter, rawValue: null, attemptedValue: null);
+                modelStateError.MarkFieldValid(parameter);
+                modelStateError.AddModelError("MyForm.MyField", errorMessage2);
```
I suspect this case succeeds without your `ValidationHelpers` fix.

In addition, this data includes a naming conflict between the parameter and the property in `FormMetadata`. Please test cases without that conflict.

natemcmaster committed:
Remove obsolete publishing targets and tasks (#7248)

pakrym committed:
Use SetPlatform in native project references (#7260)

May fix https://github.com/aspnet/AspNetCore/issues/7250 

We were double building native projects in x86 and win32 platforms

pakrym committed:
Reenable tests that are though to be fixed (#7254)

ReaderThrowsCancelledException might be fixed by https://github.com/aspnet/AspNetCore/pull/6862
DoesNotStartIfDisabled was fixed by https://github.com/aspnet/AspNetCore/pull/7068

Also remove some extreme diagnostic features added in https://github.com/aspnet/AspNetCore/pull/7068

ryanbrandenburg committed:
Handle PermanentRedirects in PrerenderTagHelpers (#7179)

Handle PermanentRedirects in PrerenderTagHelpers

pranavkm committed:
Add a TypeForward for JsonResult
Fixes https://github.com/aspnet/AspNetCore/issues/7220

pranavkm committed:
Fix test

pranavkm committed:
Add a TypeForward for JsonResult (#7278)

* Add a TypeForward for JsonResult
Fixes https://github.com/aspnet/AspNetCore/issues/7220

Tratcher committed:
Replace JObject with JsonDocument in Authentication (#7105)

halter73 committed:
Auto-set "Content-Length: 0" on 205 responses (#7205)

dotnet-maestro[bot] committed:
Update dependencies from 5 repositories (#7132)

* Update dependencies from https://github.com/aspnet/AspNetCore-Tooling build 20190129.2

This change updates the following dependencies
- Microsoft.AspNetCore.Mvc.Razor.Extensions - 3.0.0-preview-19079-02
- Microsoft.NET.Sdk.Razor - 3.0.0-preview-19079-02
- Microsoft.CodeAnalysis.Razor - 3.0.0-preview-19079-02
- Microsoft.AspNetCore.Razor.Language - 3.0.0-preview-19079-02

* Update dependencies from https://github.com/dotnet/corefx build 20190129.11

This change updates the following dependencies
- System.ComponentModel.Annotations - 4.6.0-preview.19079.11
- Microsoft.CSharp - 4.6.0-preview.19079.11
- Microsoft.Win32.Registry - 4.6.0-preview.19079.11
- System.Runtime.CompilerServices.Unsafe - 4.6.0-preview.19079.11
- System.Security.Cryptography.Cng - 4.6.0-preview.19079.11
- System.Reflection.Metadata - 1.7.0-preview.19079.11
- System.Security.Cryptography.Pkcs - 4.6.0-preview.19079.11
- System.Security.Cryptography.Xml - 4.6.0-preview.19079.11
- System.Security.Permissions - 4.6.0-preview.19079.11
- System.Security.Principal.Windows - 4.6.0-preview.19079.11
- System.ServiceProcess.ServiceController - 4.6.0-preview.19079.11
- System.Text.Encodings.Web - 4.6.0-preview.19079.11
- System.Threading.Channels - 4.6.0-preview.19079.11
- System.Data.SqlClient - 4.7.0-preview.19079.11
- System.Diagnostics.DiagnosticSource - 4.6.0-preview.19079.11
- System.Diagnostics.EventLog - 4.6.0-preview.19079.11
- System.Net.Http.WinHttpHandler - 4.6.0-preview.19079.11
- System.Net.WebSockets.WebSocketProtocol - 4.6.0-preview.19079.11
- System.Numerics.Vectors - 4.6.0-preview.19079.11
- System.IO.Pipelines - 4.6.0-preview.19079.11

* Update dependencies from https://github.com/dotnet/core-setup build 20190129.4

This change updates the following dependencies
- Microsoft.DotNet.PlatformAbstractions - 3.0.0-preview-27329-4
- Microsoft.Extensions.DependencyModel - 3.0.0-preview-27329-4
- Microsoft.NETCore.App - 3.0.0-preview-27329-4

* Update dependencies from https://github.com/aspnet/EntityFrameworkCore build 20190129.4

This change updates the following dependencies
- dotnet-ef - 3.0.0-preview.19079.4
- Microsoft.EntityFrameworkCore - 3.0.0-preview.19079.4
- Microsoft.EntityFrameworkCore.Tools - 3.0.0-preview.19079.4
- Microsoft.EntityFrameworkCore.InMemory - 3.0.0-preview.19079.4
- Microsoft.EntityFrameworkCore.Relational - 3.0.0-preview.19079.4
- Microsoft.EntityFrameworkCore.Sqlite - 3.0.0-preview.19079.4
- Microsoft.EntityFrameworkCore.SqlServer - 3.0.0-preview.19079.4

* Update dependencies from https://github.com/aspnet/AspNetCore-Tooling build 20190130.2

This change updates the following dependencies
- Microsoft.AspNetCore.Mvc.Razor.Extensions - 3.0.0-preview-19080-02
- Microsoft.NET.Sdk.Razor - 3.0.0-preview-19080-02
- Microsoft.CodeAnalysis.Razor - 3.0.0-preview-19080-02
- Microsoft.AspNetCore.Razor.Language - 3.0.0-preview-19080-02

* Update dependencies from https://github.com/dotnet/corefx build 20190130.5

This change updates the following dependencies
- System.ComponentModel.Annotations - 4.6.0-preview.19080.5
- Microsoft.CSharp - 4.6.0-preview.19080.5
- Microsoft.Win32.Registry - 4.6.0-preview.19080.5
- System.Runtime.CompilerServices.Unsafe - 4.6.0-preview.19080.5
- System.Security.Cryptography.Cng - 4.6.0-preview.19080.5
- System.Reflection.Metadata - 1.7.0-preview.19080.5
- System.Security.Cryptography.Pkcs - 4.6.0-preview.19080.5
- System.Security.Cryptography.Xml - 4.6.0-preview.19080.5
- System.Security.Permissions - 4.6.0-preview.19080.5
- System.Security.Principal.Windows - 4.6.0-preview.19080.5
- System.ServiceProcess.ServiceController - 4.6.0-preview.19080.5
- System.Text.Encodings.Web - 4.6.0-preview.19080.5
- System.Threading.Channels - 4.6.0-preview.19080.5
- System.Data.SqlClient - 4.7.0-preview.19080.5
- System.Diagnostics.DiagnosticSource - 4.6.0-preview.19080.5
- System.Diagnostics.EventLog - 4.6.0-preview.19080.5
- System.Net.Http.WinHttpHandler - 4.6.0-preview.19080.5
- System.Net.WebSockets.WebSocketProtocol - 4.6.0-preview.19080.5
- System.Numerics.Vectors - 4.6.0-preview.19080.5
- System.IO.Pipelines - 4.6.0-preview.19080.5

* Update dependencies from https://github.com/dotnet/core-setup build 20190130.4

This change updates the following dependencies
- Microsoft.DotNet.PlatformAbstractions - 3.0.0-preview-27330-4
- Microsoft.Extensions.DependencyModel - 3.0.0-preview-27330-4
- Microsoft.NETCore.App - 3.0.0-preview-27330-4

* Update dependencies from https://github.com/aspnet/EntityFrameworkCore build 20190130.1

This change updates the following dependencies
- dotnet-ef - 3.0.0-preview.19080.1
- Microsoft.EntityFrameworkCore - 3.0.0-preview.19080.1
- Microsoft.EntityFrameworkCore.Tools - 3.0.0-preview.19080.1
- Microsoft.EntityFrameworkCore.InMemory - 3.0.0-preview.19080.1
- Microsoft.EntityFrameworkCore.Relational - 3.0.0-preview.19080.1
- Microsoft.EntityFrameworkCore.Sqlite - 3.0.0-preview.19080.1
- Microsoft.EntityFrameworkCore.SqlServer - 3.0.0-preview.19080.1

* Update dependencies from https://github.com/aspnet/Extensions build 20190129.8

This change updates the following dependencies
- Internal.AspNetCore.Analyzers - 3.0.0-preview.19079.8
- Microsoft.Extensions.WebEncoders - 3.0.0-preview.19079.8
- Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Diagnostics.HealthChecks - 3.0.0-preview.19079.8
- Microsoft.Extensions.DiagnosticAdapter - 3.0.0-preview.19079.8
- Microsoft.Extensions.DependencyInjection.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.DependencyInjection - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.Xml - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.UserSecrets - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.KeyPerFile - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.Json - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.Ini - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.FileExtensions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.EnvironmentVariables - 3.0.0-preview.19079.8
- Microsoft.Extensions.FileProviders.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.CommandLine - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.AzureKeyVault - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration - 3.0.0-preview.19079.8
- Microsoft.Extensions.CommandLineUtils.Sources - 3.0.0-preview.19079.8
- Microsoft.Extensions.Caching.StackExchangeRedis - 3.0.0-preview.19079.8
- Microsoft.Extensions.Caching.SqlServer - 3.0.0-preview.19079.8
- Microsoft.Extensions.Caching.Memory - 3.0.0-preview.19079.8
- Microsoft.Extensions.Caching.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.ActivatorUtilities.Sources - 3.0.0-preview.19079.8
- Microsoft.AspNetCore.Testing - 3.0.0-preview.19079.8
- Microsoft.AspNetCore.BenchmarkRunner.Sources - 3.0.0-preview.19079.8
- Microsoft.AspNetCore.Analyzer.Testing - 3.0.0-preview.19079.8
- Microsoft.Extensions.Configuration.Binder - 3.0.0-preview.19079.8
- Microsoft.Extensions.FileProviders.Composite - 3.0.0-preview.19079.8
- Microsoft.Extensions.FileProviders.Embedded - 3.0.0-preview.19079.8
- Microsoft.Extensions.FileProviders.Physical - 3.0.0-preview.19079.8
- Microsoft.Extensions.TypeNameHelper.Sources - 3.0.0-preview.19079.8
- Microsoft.Extensions.Primitives - 3.0.0-preview.19079.8
- Microsoft.Extensions.ParameterDefaultValue.Sources - 3.0.0-preview.19079.8
- Microsoft.Extensions.Options.DataAnnotations - 3.0.0-preview.19079.8
- Microsoft.Extensions.Options.ConfigurationExtensions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Options - 3.0.0-preview.19079.8
- Microsoft.Extensions.ObjectPool - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.TraceSource - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.Testing - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.EventSource - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.Debug - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.Console - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.Configuration - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.AzureAppServices - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Logging - 3.0.0-preview.19079.8
- Microsoft.Extensions.Localization.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Localization - 3.0.0-preview.19079.8
- Microsoft.Extensions.Http - 3.0.0-preview.19079.8
- Microsoft.Extensions.Hosting.Abstractions - 3.0.0-preview.19079.8
- Microsoft.Extensions.Hosting - 3.0.0-preview.19079.8
- Microsoft.Extensions.HostFactoryResolver.Sources - 3.0.0-preview.19079.8
- Microsoft.Extensions.HashCodeCombiner.Sources - 3.0.0-preview.19079.8
- Microsoft.Extensions.FileSystemGlobbing - 3.0.0-preview.19079.8
- Microsoft.JSInterop - 3.0.0-preview.19079.8

* Add json.net reference to UI package directly

* fix new package

* latest package

* Update dependencies from https://github.com/aspnet/AspNetCore-Tooling build 20190131.3

This change updates the following dependencies
- Microsoft.AspNetCore.Mvc.Razor.Extensions - 3.0.0-preview-19081-03
- Microsoft.NET.Sdk.Razor - 3.0.0-preview-19081-03
- Microsoft.CodeAnalysis.Razor - 3.0.0-preview-19081-03
- Microsoft.AspNetCore.Razor.Language - 3.0.0-preview-19081-03

* Update Versions.props

* Update Versions.props

* Update dependencies from https://github.com/aspnet/AspNetCore-Tooling build 20190201.4

This change updates the following dependencies
- Microsoft.AspNetCore.Mvc.Razor.Extensions - 3.0.0-preview-19101-04
- Microsoft.NET.Sdk.Razor - 3.0.0-preview-19101-04
- Microsoft.CodeAnalysis.Razor - 3.0.0-preview-19101-04
- Microsoft.AspNetCore.Razor.Language - 3.0.0-preview-19101-04

* Update dependencies from https://github.com/aspnet/AspNetCore-Tooling build 20190203.1

This change updates the following dependencies
- Microsoft.AspNetCore.Mvc.Razor.Extensions - 3.0.0-preview-19103-01
- Microsoft.NET.Sdk.Razor - 3.0.0-preview-19103-01
- Microsoft.CodeAnalysis.Razor - 3.0.0-preview-19103-01
- Microsoft.AspNetCore.Razor.Language - 3.0.0-preview-19103-01

* Update Versions.props

* Update BasicWebSite.Home.CSharp7View.html

* Update dependencies from https://github.com/aspnet/AspNetCore-Tooling build 20190204.4

This change updates the following dependencies
- Microsoft.AspNetCore.Mvc.Razor.Extensions - 3.0.0-preview-19104-04
- Microsoft.NET.Sdk.Razor - 3.0.0-preview-19104-04
- Microsoft.CodeAnalysis.Razor - 3.0.0-preview-19104-04
- Microsoft.AspNetCore.Razor.Language - 3.0.0-preview-19104-04

* Update dependencies from https://github.com/dotnet/core-setup build 20190204.9

This change updates the following dependencies
- Microsoft.DotNet.PlatformAbstractions - 3.0.0-preview-27404-9
- Microsoft.Extensions.DependencyModel - 3.0.0-preview-27404-9
- Microsoft.NETCore.App - 3.0.0-preview-27404-9

* Update dependencies from https://github.com/aspnet/Extensions build 20190205.3

This change updates the following dependencies
- Internal.AspNetCore.Analyzers - 3.0.0-preview.19105.3
- Microsoft.JSInterop - 3.0.0-preview.19105.3
- Microsoft.Extensions.FileProviders.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Diagnostics.HealthChecks - 3.0.0-preview.19105.3
- Microsoft.Extensions.DiagnosticAdapter - 3.0.0-preview.19105.3
- Microsoft.Extensions.DependencyInjection.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.DependencyInjection - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.Xml - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.UserSecrets - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.KeyPerFile - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.Json - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.Ini - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.FileExtensions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.EnvironmentVariables - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.CommandLine - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.Binder - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.AzureKeyVault - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Configuration - 3.0.0-preview.19105.3
- Microsoft.Extensions.CommandLineUtils.Sources - 3.0.0-preview.19105.3
- Microsoft.Extensions.Caching.StackExchangeRedis - 3.0.0-preview.19105.3
- Microsoft.Extensions.Caching.SqlServer - 3.0.0-preview.19105.3
- Microsoft.Extensions.Caching.Memory - 3.0.0-preview.19105.3
- Microsoft.Extensions.Caching.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.ActivatorUtilities.Sources - 3.0.0-preview.19105.3
- Microsoft.AspNetCore.Testing - 3.0.0-preview.19105.3
- Microsoft.AspNetCore.BenchmarkRunner.Sources - 3.0.0-preview.19105.3
- Microsoft.AspNetCore.Analyzer.Testing - 3.0.0-preview.19105.3
- Microsoft.Extensions.FileProviders.Composite - 3.0.0-preview.19105.3
- Microsoft.Extensions.FileProviders.Embedded - 3.0.0-preview.19105.3
- Microsoft.Extensions.FileSystemGlobbing - 3.0.0-preview.19105.3
- Microsoft.Extensions.WebEncoders - 3.0.0-preview.19105.3
- Microsoft.Extensions.ValueStopwatch.Sources - 3.0.0-preview.19105.3
- Microsoft.Extensions.TypeNameHelper.Sources - 3.0.0-preview.19105.3
- Microsoft.Extensions.Primitives - 3.0.0-preview.19105.3
- Microsoft.Extensions.ParameterDefaultValue.Sources - 3.0.0-preview.19105.3
- Microsoft.Extensions.Options.DataAnnotations - 3.0.0-preview.19105.3
- Microsoft.Extensions.Options.ConfigurationExtensions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Options - 3.0.0-preview.19105.3
- Microsoft.Extensions.ObjectPool - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.TraceSource - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.Testing - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.EventSource - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.Debug - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.Console - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.Configuration - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.AzureAppServices - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Logging - 3.0.0-preview.19105.3
- Microsoft.Extensions.Localization.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Localization - 3.0.0-preview.19105.3
- Microsoft.Extensions.Http - 3.0.0-preview.19105.3
- Microsoft.Extensions.Hosting.Abstractions - 3.0.0-preview.19105.3
- Microsoft.Extensions.Hosting - 3.0.0-preview.19105.3
- Microsoft.Extensions.HostFactoryResolver.Sources - 3.0.0-preview.19105.3
- Microsoft.Extensions.HashCodeCombiner.Sources - 3.0.0-preview.19105.3
- Microsoft.Extensions.FileProviders.Physical - 3.0.0-preview.19105.3

* Update Versions.props

* Update Versions.props

Commit:
Updating the DotNetLogo bitmaps' purple color to .Net core's latest branding.

ryanbrandenburg committed:
Bootstrap 3.4.0 (#7201)

Move bootstrap 3.3 to 3.4 in samples

pakrym committed:
Add support for suspending ANCM startup (#7252)

Uses the `ASPNETCORE_STARTUP_SUSPEND_EVENT` environment variable to get an event name.

Fixes: https://github.com/aspnet/AspNetCore/issues/6972

pakrym committed:
Add DisposeAsync support to WebHost and RequestServices (#7091)

Tratcher committed:
Re-enable h2spec tests with more diagnostics Internal/#1720 (#7259)

Commit:
Updating per code review...

joeloff committed:
Merge pull request #7287 from johnbeisner/UpdateColor

Updating the DotNetLogo bitmaps' purple color to .Net core's latest branding

natemcmaster committed:
Install the .NET Core SDK into the repo root instead of UserProfile and 'install' copy of AspNetCore shared framework (#7293)

This is required to workaround several limitations in the way the .NET Core SDK finds shared frameworks and targeting packs. It allow tests to use shared frameworks and targeting packs.

It also matches the patterns established in other aspnet and dotnet repos. This should reduce the friction required to adopt Arcade SDK.

## Changes

* This moves the default location of the .NET Core SDK installation into `$repoRoot/.dotnet`. This location was already in use for CI builds. 
* Update the build step for Microsoft.AspNetCore.App to install the shared framework into the local copy of the .NET Core SDK

## Recommendations

* Use the "startvs.cmd" script to launch Visual Studio. This will set required environment variables to make VS happier than if you just double click the .sln file.
* Use "activate.sh/ps1" if you want to run `dotnet build`, `dotnet test` and other dotnet commands. These will set required environment variables, including PATH.
* I recommend removing %USERPROFILE%/.dotnet to your PATH variable if you had added it manually before. This will no longer match what build tools will install.
* `git clean -xfd -e .dotnet/` preserves the folder so you don’t have to re-download the SDK again.

natemcmaster committed:
Fix discrepancy between Version.Details.xml and Versions.props (#7314)

* Fix discrepancy between Version.Details.xml and Versions.props
* Add step to CodeCheck to ensure Version.Details.xml and Versions.props match

pakrym committed:
Use Win32 as native platform (#7309)

dougbu committed:
Add link to ASP.NET API Versioning project

pakrym committed:
Merge branch 'release/2.2'

BrennanConroy committed:
SignalR C++ tests passing and part of build (#7160)

pakrym committed:
Merge pull request #7317 from dotnet-maestro-bot/merge/release/2.2-to-master

[automated] Merge branch 'release/2.2' => 'master'

javiercn committed:
[Components] [Fixes #6953, #7226]

* Moves the Synchronization context from the remote renderer to the base renderer.
* Removes all the locking from the base renderer.

BrennanConroy committed:
Use the 'correct' windows platform version (#7338)

jkotalik committed:
Remove misleading websocket error log (#7249)

pakrym committed:
Remove ANCMV1 from everywhere (#7334)

jmazouri committed:
Fix typo in HttpSysOptions.cs (#7207)

`infiniate` => `infinite` in ArgumentOutOfRangeException message.

HaoK committed:
Run helix tests on master (#6728)

pakrym committed:
Remove Microsoft.AspNetCore.All from site extension description (#7335)

kishanAnem committed:
Merge branch 'master' of https://github.com/kishanAnem/AspNetCore

kishanAnem committed:
ValidationSummary doesn't display property errors if ModelState and M… #7221

Hi @dougbu

I've removed test which is having naming conflicts.

kishanAnem commented:
@SteveSandersonMS, @Tratcher, @natemcmaster and @jkotalik.
Sorry I did commit fa55c55. @dougbu  is reviewing these changes.

halter73 committed:
Fix flaky CanUpgradeRequestWithConnectionKeepAliveUpgradeHeader test (#7323)

* Randomize MockSystemClock

dougbu commented:
I can no longer read this. Could you please rebase your fork instead of merging in our `master`?

natemcmaster commented:
Can you rebase the PR on master? GitHub is showing 10k changes on hundreds of files.

![image](https://user-images.githubusercontent.com/2696087/52438955-55f18800-2acf-11e9-852f-77d0f7472fab.png)

kishanAnem commented:
I'll do rebase fork.

Commit:
Unset reserved bit when setting 31 bit uint values (#7339)

* Unset reserved bit when setting 31 bit uint values

* Add FrameWriter tests

* Add overload

BrennanConroy committed:
C++ client: Skip flaky tests (#7358)

AArnott committed:
Remove extraneous call to GetMemory in BufferWriter (#7354)

The BufferWriter called GetMemory(count) and dropped the result, only to then call GetSpan(). This moves the count argument to GetSpan, and drops the call to GetMemory.

halter73 committed:
Merge branch 'release/2.2'

HaoK committed:
Regenerate the stamp once if there's a token collision in test (#7364)

HaoK committed:
Reorganize helix files (#7353)

AArnott committed:
Add xml doc comments for BufferWriter struct (#7350)

dukhanov committed:
Add webworker support to SignalR JS client (#7058)

* Added Platform utils to detect platform type
* Added additional build for WebWorker
* Changed env param from webworker to platform to make ability to specify platform to the build script
* Updated the readme file with SignalR WebWorker instructions

natemcmaster committed:
Produce a package with just reference assemblies for AspNetCore.App (#7355)

davidfowl committed:
Handle IAsyncDisposable scoped objects (#7343)

- We make a scope today around hub invocations, with IAsyncDisposable now implemented in the DI container, we need to support IServiceScope being IAsyncDisposable and IDisposable

JamesNK committed:
Small fixes to SignalR service extension methods (#7375)

kishanAnem committed:
Merge branch 'master' of https://github.com/kishanAnem/AspNetCore

kishanAnem commented:
Hi @dougbu 

I've rebased my fork following way:
git remote add Mvc https://github.com/aspnet/AspNetCore.git
git pull --rebase upstream master
git push --force-with-lease origin master
git checkout master
git reset --hard Mvc/master

commits 
bfe6a7b
cc1b0e3
fa55c55

Thanks @dougbu. I've tried rebase my fork. it didn't work. It is going bad. If I do some more. 
Please close PR, if it is useless.

dougbu commented:
@kishanAnem I suspect the issue here is you've been merging `upstream/master` into your `origin/master` branch. To get things back on track (assuming you have one commit to preserve)
1. `git log --author kishanAnem`
2. `git reset --hard {commit containing the fix}`
3. `git branch -b fix.7221 master`
4. `git checkout master`
5. `git log`
6. `git reset --hard {commit before your fix}` // Should be back to when you created the fork
7. `git fetch upstream`
8. `git merge --ff-only upstream/master`
9. `git push --force origin`
10. `git checkout fix.7221`
11. `git rebase` // Remember previous commit in case it gets lost (may need `git cherry-pick {commit}`)
12. `git push origin`
13. Click "Edit" in this PR and change the source branch from `master` to `fix.7221`

Going forward, keep `master` up-to-date and use other branches for proposals. Rebase the other branches as needed.

kishanAnem commented:
@dougbu I will do as you said.
